File: merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key= lambda x: x[0])
        mergedIntervals = [intervals[0]]
        for interval in intervals[1:]:
            if mergedIntervals[-1][1] < interval[0]:
                mergedIntervals.append(interval)
            else:
                mergedIntervals[-1][1] = max(mergedIntervals[-1][1], interval[1])
        return mergedIntervals

File: c++/merge-intervals.cc
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> mergedIntervals;
        mergedIntervals.push_back(intervals[0]);
        for (int i = 1; i < intervals.size(); ++i) {
            auto& lastElem = mergedIntervals[mergedIntervals.size() - 1];
            if (intervals[i][0] <= lastElem[1]) {
                lastElem[1] = max(intervals[i][1], lastElem[1]);
            } else {
                mergedIntervals.push_back(intervals[i]);
            }
        }
        return mergedIntervals;
    }
};